application/hubspot/v0/config/tasks.json

```diff
@@ -0,0 +1,1149 @@
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_CREATE_DEAL": {
+    "instillShortDescription": "Create new deal",
+    "input": {
+      "description": "Deal information",
+      "instillUIOrder": 0,
+      "properties": {
+        "contact-id-or-email": {
+          "description": "Use to associate contact with deal. If the input has @, it will use email, otherwise, it will use contact id.",
+          "$ref": "#/$defs/common/contact-id-or-email"
+        },
+        "owner-id": {
+          "$ref": "#/$defs/common/owner-id",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "deal-name": {
+          "$ref": "#/$defs/deal/deal-name",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "pipeline": {
+          "$ref": "#/$defs/deal/pipeline",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "deal-stage": {
+          "$ref": "#/$defs/deal/deal-stage",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "amount": {
+          "$ref": "#/$defs/deal/amount",
+          "instillAcceptFormats": ["number"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "deal-type": {
+          "$ref": "#/$defs/deal/deal-type",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        },
+        "close-date": {
+          "$ref": "#/$defs/deal/close-date",
+          "instillAcceptFormats": ["string"],
+          "instillUIMultiline": true,
+          "instillUpstreamTypes": ["value", "reference", "template"]
+        }
+      },
+      "required": ["deal-name", "pipeline", "deal-stage"],
+      "title": "Input",
+      "type": "object"
+    },
+    "output": {
+      "description": "Obtain deal ID",
+      "instillUIOrder": 0,
+      "properties": {
+        "deal-id": {
+          "description": "Deal ID",
+          "instillUIOrder": 0,
+          "required": [],
+          "title": "Deal ID",
+          "type": "string",
+          "instillFormat": "string"
+        }
+      },
+      "required": ["deal-id"],
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_GET_COMPANY": {
+    "instillShortDescription": "Get company information using company ID",
+    "input": {
+      "description": "Input company ID",
+      "instillUIOrder": 0,
+      "properties": {
+        "company-id": {
```

Comment:
> question
Similar with the question for contact-id.
Not so sure how user get this company-id. And, also not so sure in what stories it will be used.

Company-id is obtainable using "Retrieve Association" task. Using contact id, user can get all objects id (companies-id, deals-id, threads-id, tickets-id) associated with that contact id. 
I implemented the company task in order for user in VDP to easily get information about the client's company. Maybe the user (VDP user) wants to get the company's email and use the email component to write an email.

> Share
When there are multiple APIs, I usually draw an image like this to help me clarify the usage of each API.
Each API should their own story usage when we call them.

thanks for sharing! I will probably do that as well for future components so that it is easier to review